backend/src/main/java/com/google/capstone/UserUploadHandler.java

```diff
@@ -0,0 +1,104 @@
+ * Note: must be updated before deployment to redirect to proper frontend server
+ */
+@WebServlet("/api/v1/uploadImgs")
+public class UserUploadHandler extends HttpServlet{
+ 
+  public static final String GCS_BUCKET_NAME = "upload-imgs";
+ 
+  private static final BlobstoreService blobstore = BlobstoreServiceFactory.getBlobstoreService();
+ 
+  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+ 
+  private static final ImagesService images = ImagesServiceFactory.getImagesService();
+ 
+  /**
+   * Creates an upload URL to send the image to in GCS.
+   * is used in the fetch that returns URL to action in form
+   */  
+  @Override
+  protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UploadOptions opts = UploadOptions.Builder.withGoogleStorageBucketName(GCS_BUCKET_NAME);
+    String uploadUrl = blobstore.createUploadUrl("/api/v1/uploadImgs", opts);
+ 
+    response.setContentType("text/html");
+    response.getOutputStream().println(uploadUrl);
+  }
+  
+  /**
+   * Gets image information from form and stores in Datastore 
+   * Also creates and store blobKey and serveUrl
+   */
+  @Override
+  protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+      Map<String, List<BlobKey>> blobs = blobstore.getUploads(request);
+      List<BlobKey> blobKeys = blobs.get("selectedFile"); 
+ 
+      //checks to make sure image was added to create blobkey
+      if (blobKeys == null || blobKeys.isEmpty()) {
+        /* NOTE: add error page here, no file selected!!
+           Will create better error handling in future update :) 
+           Currently redirect to museumGallery for testing reasons */
+        response.sendRedirect("https://3001-ba659410-163c-49e0-b45f-c22c5b2dc8b5.us-central1.cloudshell.dev/museum-gallery");
+        return;
+      }
+ 
+      //gets info from form, entered by user
+      String artistName = getParameter(request, "artistName", "");
+      String artTitle = getParameter(request, "artTitle", "");
+      String description = getParameter(request, "description", "");
+      long timestamp = System.currentTimeMillis();
+ 
+      //ImageInformation entity to be added into datastore
+      Entity mssgEntity = new Entity("ImageInformation");
+      mssgEntity.setProperty("artistName", artistName);
+      mssgEntity.setProperty("artTitle", artTitle);
+      mssgEntity.setProperty("description", description);
+      mssgEntity.setProperty("timestamp", timestamp);
+ 
+      String blobKey = blobKeys.get(0).getKeyString();
+      mssgEntity.setProperty("blobKey", blobKey);
+      mssgEntity.setProperty("url", "/api/v1/get-blob?blob-key=" + blobKey);
+      
+      datastore.put(mssgEntity);  
+      
+      //deployment link
+      response.sendRedirect("https://igunda-isangimino-nstroupe.uc.r.appspot.com/user-uploads-gallery");
```

Comment:
Using absolute path doesn't seems right here. Check https://www.codejava.net/java-ee/servlet/how-to-send-redirect-from-java-servlet for how to use relative path. If you still have problem with redirecting, I am happy to debug with you.